### ISSUE TITLE
[insiders] Code block annotation alignment for comment lines

### ISSUE THREAD

**Issue by kcgthb** [squidfunk/mkdocs-material#3145]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Code block annotations don't appear to be properly positioned on comment lines.

### Expected behaviour

Properly aligned annotation mark.

### Actual behaviour

The following Markdown:
      
    # Annotation test
      
      ```shell
      #!/bin/bash
      #
      # this is a multi-line comment
      # that goes on and on # (1)
      # and ends here
      
      echo "not a comment" # (2)
      echo "the end"
      ```
      
      1. first annotation
      2. second annotation
    
generates the following code block:
![image](https://user-images.githubusercontent.com/186807/138792734-32147e90-3875-43cd-b058-87a4a79be049.png)

* Annotation (1) is made on a commented line, and the marker is inserted as a new line in the block.
* Annotation (2) is made on a regular, non-comment line, and the marker properly aligned.

### Steps to reproduce

Activate code block annotations in Insiders, and generated a code block with an annotation on a language-commented line.

### Package versions

- Python: Python 3.9.7
- MkDocs: mkdocs, version 1.2.3
- Material: Version: 7.3.4+insiders.3.1.4

### Configuration

```yaml
theme:
  features:
    - content.code.annotate
```

### System information

- Operating system: Linux
- Browser: Chrome

**Comment by squidfunk** [comment]
Fixed in [`8ebee733d`](https://github.com/squidfunk/mkdocs-material-insiders/commit/8ebee733da268abc5296962647a4e9b2f3e8b12b).

<img width="714" alt="Bildschirmfoto 2021-11-07 um 13 44 08" src="https://user-images.githubusercontent.com/932156/140645382-931731d3-a8a9-4131-85f8-80568008e4a7.png">

Before this commit, a comment was entirely replaced when a code annotation was found in it. Now, comments are scanned for code annotations and the respective text nodes are split, only replacing the actual annotations. This allows for putting annotations inside comments (see bubble 1 and 2)

However, it also means that now the characters opening the comment, i.e. `#`, `//` etc. are kept (see bubble 3). Personally, I like it, but other users maybe don't. Thus, we might need a way to signal via annotation syntax that an annotation should replace the comments in its entirety, maybe something like `(1)!`. This however doesn't look as nice when rendered without JavaScript (with annotations not being available). Another way would be to detect whether a comment does solely contain a code annotation, but then we would also need some mechanism to detect what is a control character and what not, so it's probably better to let the user decide. Any mechanism must work with all languages.

I'm interested in feedback before pushing this out as part of a new release.

**Comment by squidfunk** [comment]
I had to reissue the latest release (my fault), so this has been released as part of 7.3.6+insiders-3.2.2. I'll let this open until we find out how to build a sensible custom report from this. Sorry for any inconvenience this might be causing.
